include/klee/Internal/Module/KModule.h

```diff
@@ -107,14 +109,30 @@ namespace klee {
     void addInternalFunction(const char* functionName);
 
   public:
-    KModule(llvm::Module *_module);
-    ~KModule();
+    KModule() = default;
 
-    /// Initialize local data structures.
+    /// Optimise and prepare module such that KLEE can execute it
     //
     // FIXME: ihandler should not be here
-    void prepare(const Interpreter::ModuleOptions &opts, 
-                 InterpreterHandler *ihandler);
+    void optimiseAndPrepare(const Interpreter::ModuleOptions &opts,
+                            InterpreterHandler *ihandler,
+                            llvm::ArrayRef<const char *>);
+
+    /// Manifests the generated module (e.g. assembly.ll, output.bc) and
+    /// prepares KModule
+    ///
+    /// @param ih
+    /// @param forceSourceOutput true if assembly.ll should be created
+    void manifest(InterpreterHandler *ih, bool forceSourceOutput);
+
+    /// Link provided modules together as one KLEE module
+    ///
+    // FIXME: ihandler should not be here
```

Comment:
It seems like this can be fixed now. If I'm not mistaken, the implementation of KModule::link does not access `ihandler`.